packages/flutter_tools/lib/src/macos/macos_device.dart

```diff
@@ -74,6 +77,26 @@ class MacOSDevice extends DesktopDevice {
       targetOverride: mainPath,
       verboseLogging: _logger.isVerbose,
     );
+    final String? executablePath = executablePathForDevice(package, buildInfo);
+    if (executablePath != null) {
+      final String appPath = _fileSystem.path.join(executablePath);
+      await unquarantineApp(appPath);
+    }
+  }
+
+  /// Marks the specified application as unquarantined.
+  ///
+  /// As of macOS 14 (Sonoma), macOS marks newly-compiled binaries as
+  /// quarantined, and displays a warning about running an unsigned app before
+  /// launching them. Since these binaries are compiled and run at the
+  /// developers request, the author is aware that they're unsigned, mark them
+  /// as unquarantined.
+  ///
+  /// See: https://github.com/flutter/flutter/issues/136888
+  Future<void> unquarantineApp(String applicationPath) async {
+    // Ignore exit status. If the command fails, the user can still run the app,
+    // but will get a warning popup.
+    await _processManager.run(<String>['xattr', '-d', 'com.apple.quarantine', applicationPath]);
```

Comment:
Is this process's output swallowed? Would it make sense to log it if in verbose mode in case this breaks? I don't feel strongly about this, feel free to skip

Reply:
`xattr` has no output, just an exit code, but we could log it anyway just in case it ever does.